src/pod.jl

```diff
@@ -241,3 +241,28 @@ function isk8s()
     end
     return in_kubepod
 end
+
+
+"""
+    pod_zone(pod_name::AbstractString) -> String
+
+Get the zone that the node hosting the current pod is running in.
+"""
+function pod_zone(pod_name::AbstractString)
+    node_name = readchomp(`$(kubectl()) get pod $pod_name -o jsonpath='{.spec.nodeName}'`)
+    return readchomp(`$(kubectl()) get node $node_name -o jsonpath='{.metadata.labels.topology\.kubernetes\.io/zone}'`)
+end
+
+
+"""
+    zone_node_selector!() -> Dict
+
+Add a nodeSelector to a new pod that will ensure that it runs in the same zone
+as the manager pod.
+"""
+function zone_node_selector!(pod)
+    pod_name = pod["metadata"]["labels"]["manager"]
+    zone = pod_zone(pod_name)
```

Comment:
Since `pod_name` refers to a different pod than `pod`, maybe `manager_name`?
```suggestion
    manager_name = pod["metadata"]["labels"]["manager"]
    zone = pod_zone(manager_name)
```